runtime/src/iree/hal/drivers/hip/hip_allocator.c

```diff
@@ -657,16 +709,40 @@ iree_status_t iree_hal_hip_allocator_alloc_async(
     iree_hal_buffer_t* buffer) {
   iree_hal_hip_allocator_t* allocator =
       iree_hal_hip_allocator_cast(base_allocator);
+  if (iree_hal_buffer_allocation_size(buffer) > (size_t)-1 ||
+      iree_hal_buffer_allocation_size(buffer) > (iree_host_size_t)(-1)) {
+    return iree_make_status(
+        IREE_STATUS_INVALID_ARGUMENT,
+        "buffer allocation too large for hip on this system");
```

Comment:
when hitting cases like this include the values - for example, `"buffer allocation of %" PRIdsz " too large to map into host address space with max of %" PRIhsz, iree_hal_buffer_allocation_size(buffer), IREE_HOST_SIZE_MAX)`

the goal is to not make the user come in and add breakpoints/printfs/etc - when it's cheap to do so (like here) always include additional information about what specifically caused the error